pkg/event/utils.go

```diff
@@ -0,0 +1,107 @@
+package event
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/content-services/content-sources-backend/pkg/event/schema"
+	"github.com/rs/zerolog/log"
+)
+
+var defaultTopic = "__consumer_offsets"
+
+type KafkaMessagePredicate func(msg *kafka.Message) bool
+
+type pingable interface {
+	GetMetadata(topic *string, allTopics bool, timeoutMs int) (*kafka.Metadata, error)
+}
+
+func Ping(timeout int, instances ...pingable) error {
+	for _, instance := range instances {
+		if _, err := instance.GetMetadata(&defaultTopic, false, timeout); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func Headers(keysAndValues ...string) []kafka.Header {
+	if len(keysAndValues)%2 != 0 {
+		panic(fmt.Sprintf("Odd number of parameters: %s", keysAndValues))
+	}
+
+	result := make([]kafka.Header, len(keysAndValues))
+
+	for i := 0; i < len(keysAndValues)/2; i++ {
+		result[i] = kafka.Header{
+			Key:   keysAndValues[i*2],
+			Value: []byte(keysAndValues[(i*2)+1]),
+		}
+	}
+
+	return result
+}
+
+func GetHeader(msg *kafka.Message, key string) (string, error) {
+	for _, value := range msg.Headers {
+		if value.Key == key {
+			return string(value.Value), nil
+		}
+	}
+
+	return "", fmt.Errorf("Header not found: %s", key)
+}
+
+func FilterByHeaderPredicate(ctx context.Context, header string, filterVals ...string) KafkaMessagePredicate {
```

Comment:
Is this still being used somewhere? I can't find it being called.

Reply:
I have deleted it; not used.